packages/hardhat/contracts/QuestFactory.sol

```diff
@@ -2,22 +2,44 @@
 pragma solidity ^0.8.1;
 
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./libraries/Deposit.sol";
+import "./libraries/Models.sol";
 import "./Quest.sol";
 
-contract QuestFactory {
+contract QuestFactory is Ownable {
+    using SafeERC20 for IERC20;
+    using DepositLib for Models.Deposit;
+
     address public aragonGovernAddress;
+    Models.Deposit public deposit;
 
     event QuestCreated(
         address questAddress,
         string questTitle,
         bytes questDetailsRef,
         address rewardTokenAddress,
         uint256 expireTime,
-        uint256 creationTime
+        uint256 creationTime,
+        address depositToken,
+        uint256 depositAmount
     );
 
-    constructor(address _aragonGovernAddress) {
+    event DepositChanged(uint256 timestamp, address token, uint256 amount);
+
+    constructor(
+        address _aragonGovernAddress,
+        IERC20 _depositToken,
+        uint256 _depositAmount
+    ) {
         aragonGovernAddress = _aragonGovernAddress;
+        setDeposit(_depositToken, _depositAmount);
+    }
+
+    function setDeposit(IERC20 token, uint256 amount) public onlyOwner {
```

Comment:
We can consider using `using SafeMath for uint256;` for the uint256 amounts, depositamount and others variables. To avoid overflow and underflow.